src/components/Proposal/Create/SchemeForms/CreateGenericMultiCallProposal.tsx

```diff
@@ -0,0 +1,502 @@
+  handleClose: () => any;
+  scheme: ISchemeState;
+  whitelistedContracts: Array<string>;
+}
+
+interface IDispatchProps {
+  createProposal: typeof createProposal;
+  showNotification: typeof showNotification;
+}
+
+type AddContractError = "NOT_VALID_ADDRESS" | "CONTRACT_EXIST" | "ABI_DATA_ERROR" | "";
+
+interface IAddContractStatus {
+  error: AddContractError;
+  message: string;
+}
+
+interface IStateProps {
+  loading: boolean;
+  tags: Array<string>;
+  addContractStatus: IAddContractStatus;
+  whitelistedContracts: Array<string>;
+  userContracts: Array<string>;
+}
+
+type IProps = IExternalProps & IDispatchProps;
+
+const mapDispatchToProps = {
+  createProposal,
+  showNotification,
+};
+
+interface IContract {
+  address: string; // Contract address
+  value: number; // Token to send with the proposal
+  abi: any; // Contract ABI data
+  methods: any; // ABI write methods
+  method: string; // Selected method
+  params: any; // Method params
+  values: any; // Params values
+  callData: string; // The encoded data
+}
+
+interface IFormValues {
+  description: string;
+  title: string;
+  url: string;
+  contracts: Array<IContract>;
+  [key: string]: any;
+}
+
+class CreateGenericMultiCallScheme extends React.Component<IProps, IStateProps> {
+
+  initialFormValues: IFormValues;
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.initialFormValues = importUrlValues<IFormValues>({
+      description: "",
+      title: "",
+      url: "",
+      tags: [],
+      addContract: "",
+      userContracts: [],
+      contracts: [
+        {
+          address: "",
+          value: 0,
+          abi: [],
+          methods: [],
+          method: "",
+          params: [],
+          values: [],
+          callData: "",
+        },
+      ],
+    });
+    this.state = {
+      loading: false,
+      tags: this.initialFormValues.tags,
+      addContractStatus: { error: "", message: "" },
+      whitelistedContracts: this.props.whitelistedContracts?.map(contract => { return contract.toLowerCase(); }) ?? [],
+      userContracts: [],
+    };
+  }
+
+  public async handleSubmit(formValues: IFormValues, { setSubmitting }: any): Promise<void> {
+    if (!await enableWalletProvider({ showNotification: this.props.showNotification })) { return; }
+
+    const contractsToCall = [];
+    const callsData = [];
+    const values = [];
+
+    for (const contract of formValues.contracts) {
+      contractsToCall.push(contract.address);
+      callsData.push(contract.callData);
+      values.push(contract.value);
+    }
+
+    const proposalValues = {
+      title: formValues.title,
+      description: formValues.description,
+      contractsToCall: contractsToCall,
+      callsData: callsData,
+      values: values,
+      url: formValues.url,
+      dao: this.props.daoAvatarAddress,
+      scheme: this.props.scheme.address,
+      tags: this.state.tags,
+    };
+
+    setSubmitting(false);
```

Comment:
what this one is doing ?

Reply:
This is Forimk function to inform the form that we're no longer in a submitting process. This is a standard in all our Formik forms.